src/Context/BasicContext.php

```diff
@@ -375,18 +376,19 @@ public function stepIWaitFor($secs)
      * @param string $title
      * @return NodeElement|null
      */
-    protected function findNamedButton($title)
+    public function findNamedButton($title, ?ElementInterface $parent = null)
     {
-        $page = $this->getSession()->getPage();
+        if ($parent === null) {
+            $parent = $this->getSession()->getPage();
+        }
         // See https://mathiasbynens.be/notes/css-escapes
         $escapedTitle = addcslashes($title ?? '', '!"#$%&\'()*+,-./:;<=>?@[\]^`{|}~');
-        $matchedEl = null;
         $searches = [
             ['named', ['link_or_button', "'{$title}'"]],
             ['css', "button[data-text-alternate='{$escapedTitle}']"],
         ];
         foreach ($searches as list($type, $arg)) {
-            $buttons = $page->findAll($type, $arg);
+            $buttons = $parent->findAll($type, $arg);
```

Comment:
Needed for https://github.com/silverstripe/silverstripe-asset-admin/pull/1544#discussion_r1982665971